Core/include/Acts/Vertexing/SeedVertexFinder.ipp

```diff
@@ -0,0 +1,656 @@
+    const std::vector<Acts::SeedVertexFinder<spacepoint_t>::Triplet>& triplets)
+    const {
+  // define function f = sum over all triplets [distance from an unknown point
+  // (x_0,y_0,z_0) to the ray defined by the triplet] find minimum of "f" by
+  // partial derivations over x_0, y_0, and z_0 each derivation has parts
+  // lineary depending on x_0, y_0, and z_0 (will fill A[][3]) or to nothing
+  // (will fill B[]) solve A*(x_0,y_0,z_0) = B
+
+  Acts::Vector3 vtx = Acts::Vector3::Zero();
+  Acts::Vector3 vtx_prev{m_cfg.rMaxFar, m_cfg.rMaxFar, m_cfg.maxAbsZ};
+
+  // (start_point, direction), distance
+  std::vector<
+      std::pair<std::pair<Acts::Vector3, Acts::Vector3>, Acts::ActsScalar>>
+      triplets_with_rays;
+  triplets_with_rays.reserve(triplets.size());
+
+  for (const auto& triplet : triplets) {
+    triplets_with_rays.emplace_back(
+        std::make_pair(triplet.ray.origin(), triplet.ray.dir()), -1.);
+  }
+
+  // elements of the linear equations to solve
+  Acts::SymMatrix3 A = Acts::SymMatrix3::Identity() * 2. * triplets.size();
+  Acts::Vector3 B = Acts::Vector3::Zero();
+  for (const auto& triplet : triplets_with_rays) {
+    // use ray saved from earlier
+    const auto& start_point = triplet.first.first;
+    const auto& direction = triplet.first.second;
+
+    A -= 2. * (direction * direction.transpose());
+    B += -2. * direction * (direction.dot(start_point)) + 2. * start_point;
+  }
+
+  for (int iter = 0; iter <= m_cfg.maxIterations; iter++) {
+    // new vertex position
+    vtx = A.lu().solve(B);
+
+    Acts::Vector3 vtx_diff = vtx - vtx_prev;
+
+    if (std::sqrt(vtx_diff.dot(vtx_diff)) < m_cfg.minVtxShift) {
+      // difference between the new vertex and the old vertex is not so large
+      break;
+    }
+
+    if (iter != m_cfg.maxIterations) {
+      // is not the last iteration
+      vtx_prev = vtx;
+
+      for (auto& triplet : triplets_with_rays) {
+        const auto& start_point = triplet.first.first;
+        const auto& direction = triplet.first.second;
+
+        Acts::Vector3 vec_x_direction = (vtx - start_point).cross(direction);
+        Acts::ActsScalar distance =
+            std::sqrt(vec_x_direction.dot(vec_x_direction));
+
+        triplet.second = distance;
+      }
+
+      std::sort(triplets_with_rays.begin(), triplets_with_rays.end(),
+                [](const auto& lhs, const auto& rhs) {
+                  return lhs.second < rhs.second;
+                });
+
+      unsigned int threshold = (unsigned int)(triplets_with_rays.size() *
+                                              (1. - m_cfg.removeFraction));
+
+      for (unsigned int tr = threshold + 1; tr < triplets_with_rays.size();
+           ++tr) {
+        const auto& start_point = triplets_with_rays.at(tr).first.first;
+        const auto& direction = triplets_with_rays.at(tr).first.second;
+
+        // remove this triplet from A and B
+        A += 2. * (direction * direction.transpose());
```

Comment:
`squaredNorm`?

Reply:
Hi @andiwand, same as above, we'd like to make a matrix.

Cheers, Petr